app/main/views/auth.py

```diff
@@ -97,10 +97,10 @@ def match_idp_email(idp, email_address):
 
 
 def dept_from_idp_name(idp_name):
-    idp = [idp['name'] for idp in idp_profiles if idp_name == idp['idp_name']]
-    if idp:
-        return ' or '.join(idp)
-    return None
+    dept = [profile['name']
+            for profile in idp_profiles if idp_name == profile['idp_name']]
+    if dept:
+        return ' or '.join(dept)
```

Comment:
Arbitrary indentation level on line 101. But rather than using a list comprehension, consider using a for loop instead - list comprehensions are less readable. Also, since the callers of this function do not handle a `None` return value, the check on line 102 is redundant. Eg:
```python
dept_names = []

for profile in idp_profiles:

    if idp_name == profile['idp_name']:
        dept_names.append(profile['name'])

return ' or '.join(dept_names)
```